http_client.rst

```diff
@@ -125,24 +146,107 @@ Authentication
 ~~~~~~~~~~~~~~
 
 The HTTP client supports different authentication mechanisms. They can be
-defined globally when creating the client (to apply it to all requests) and to
-each request (which overrides any global authentication)::
+defined globally in the configuration (to apply it to all requests) and to
+each request (which overrides any global authentication).
+
+Use the ``scoped_clients`` options to create an HTTP client service with
+pre-configured options. Read more about them :ref:`below <http-scoped-clients>`:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            http_client:
+                scoped_clients:
+                    # This creates a new HTTP Client service named "example_api".
+                    # It will be autowired in arguments type-hinting "HttpClientInterface $example_api"
+                    example_api:
```

Comment:
I'm not sure this example is OK ... because it defines 3 different (and incompatible?) security-related options at the same time.

This article starts with a real and fully-working example based on the GitHub API. We could continue using that and create a scope client for GitHub which sets the `Accept` request header, which is something that GitHub encourages when using their API: https://developer.github.com/v3/#current-version

Update: when continuing the review, I saw that we have exactly that example later when "scoped clients" are explained again. I recommend instead to mention7explain scoped clients in just one place.